File: leetcode118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        initialList = [1,1]
        fullSet = [[1], [1,1]]
        if numRows == 1:
            return fullSet[:1]
        elif numRows == 2:
            return fullSet
        for x in range(numRows-2):
            addingList = [1]
            rightPointer = 1
            leftPointer = 0
            while rightPointer < len(initialList):
                addingList.append(initialList[leftPointer]+initialList[rightPointer])
                rightPointer += 1
                leftPointer += 1
            addingList.append(1)
            fullSet.append(addingList)
            initialList = addingList
        return fullSet

File: javaStuff/leetcode118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        for(int i = 0; i < numRows; i++){
            List<Integer> layer = new ArrayList<>();
            for(int j = 0; j < i + 1 ; j++) {
                if(j == 0 || j == i) {
                    layer.add(1);
                }
                else {
                    int a = result.get(i - 1).get(j - 1);
                    int b = result.get(i - 1).get(j);
                    layer.add(a + b);
                }
            }
            result.add(layer);
        }
        return result;
    }
}